src/span.js

```diff
@@ -144,6 +144,17 @@ export default class Span {
      **/
     setOperationName(operationName: string): Span {
         this._operationName = operationName;
+        // If the span has debug set, then the question of resampling
+        // after setting the operation name is irrelevant.  The span is always
+        // sampled.
```

Comment:
I don't see any reason, but it doesn't mean we should not have a strategy for handling it. What does it mean to say "user error"? We're not going to throw an exception on it.